mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,81 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).jssosn({
+      error: 'Access denied',
+    });
+  }
+  next();
+};
+
+exports.forgotPassword = (req, res) => {
+  if (!req.body) return res.status(400).json({ message: 'No request body' });
+  if (!req.body.email)
+    return res.status(400).json({ message: 'No Email in request body' });
+
+  console.log('forgot password finding user with that email');
+  const { email } = req.body;
+  console.log('signin req.body', email);
+
+  User.findOne({ email }, (err, user) => {
+    if (err || !user)
+      return res.status('401').json({
+        error: 'User with that email does not exist!',
+      });
+
+    const token = jwt.sign(
+      { _id: user._id, iss: 'NODEAPI' },
+      process.env.JWT_SECRET
+    );
+
+    const emailData = {
+      from: '',
+      to: email,
+      subject: 'Password Reset Instructions',
```

Comment:
The subject line 'Password Reset Instructions' is not used anywhere in the provided code snippet. It seems to be a part of an email template, but it's not clear how it's being used in the context of the code. It would be better to provide more context or clarify how this subject line is being utilized.